### ISSUE TITLE
[question] Usage of Excel export with CSP header

### ISSUE THREAD

**Issue by Jul13nT** [mui/mui-x#9572]
### Order ID

69724

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I'm trying to use the Excel export feature with the premium Data Grid. My application is deployed with a NGINX server which adds a CSP header with `script-src 'self';`. When using the export, the script is blocked by the CSP header because of a `Function` instruction.

I traced the cause to the library [regenerator-runtime](https://github.com/facebook/regenerator/blob/063f14ef7f01bb29830e3f17e9ef151e7a5cb2f3/packages/regenerator-runtime/runtime.js#L713-L726), used by ExcelJS. This seems to be a known issue: https://github.com/exceljs/exceljs/issues/713 and https://github.com/facebook/regenerator/issues/378#issuecomment-874296918

I use vite to build my application, I managed to "fix" the problem by adding this to the vite config:

```javascript
build: {
    rollupOptions: {
        output: {
            intro: 'window.regeneratorRuntime = undefined;',
        },
    },
},
```

My question is: have you ever encountered this problem and do you have a better solution to solve it?

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
  Binaries:
    Node: 18.16.1 - /usr/bin/node
    Yarn: 3.2.4 - /usr/bin/yarn
    npm: 9.5.1 - /usr/bin/npm
  Browsers:
    Chrome: 114.0.5735.198
```
</details>

**Comment by cherniavskii** [comment]
Closing in favor of https://github.com/mui/mui-x/issues/11153
